Tools/CISupport/ews-build/events.py

```diff
@@ -198,7 +198,7 @@ def buildFinished(self, key, build):
         builder_display_name = builder.get('description')
 
         if self.extractProperty(build, 'github.number'):
-            return self.buildFinishedGitHub(builder_display_name, build)
+            return self.buildFinishedGitHub(builder_display_name + custom_suffix, build)
```

Comment:
Modifying builder name seems weird. Can you include this information in some other manner, maybe in context (like you are doing for step)?

Reply:
The `context` acts as a dictionary key in GitHub, essentially. So if we use the same context for UAT and production, we'll be overwriting the same status.

Reply:
That means, we can use different `context` for uat and production, and github would be able to differentiate them, right?

Reply:
Right. GitHub will treat different contexts as two independent statuses.

Reply:
Looking more into it, my concern was modifying builder name here. But seems like it is eventually being assigned to context only.

It wasn't obvious that `buildFinishedGitHub` method assign builder_display_name to context variable. It's probably better not to have `context` parameter, instead just have the `builder` parameter, and let `buildFinishedGitHub` function create context from the builder appropriately, somewhat similar to `stepStartedGitHub`. That would make `stepStartedGitHub` and `buildFinishedGitHub` consistent.